### ISSUE TITLE
Bug: The plugin doesn't work on android.

### ISSUE THREAD

**Issue by YIRU69** [tadashi-aikawa/obsidian-another-quick-switcher#207]
Hi, I found a bug on this version, it don't work on andriod.

Plugin version: 9.1.1
obsidian version: 1.4.12

I just try to use recent search. I try reinstall and it also don't work. 

1. Open "Recent search", it show the recent files.
2. Use my finger to touch a file. And it can't open the file.

**Comment by tadashi-aikawa** [comment]
Thank you for your report, @YIRU69 :)

Unfortunately, I have never used Obsidian mobile 1.4.12 because that version is still unstable. When the version is released as stable, I'll survey the problem.

**Comment by kometenstaub** [comment]
I can confirm the same behaviour on iOS. 

There are no errors in the console, however.

**Comment by tadashi-aikawa** [comment]
I could confirm that it doesn't open the suggestions on the dialog on Android in the latest stable mobile version.

**Comment by tadashi-aikawa** [comment]
I found the cause.

https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/blob/master/src/ui/AnotherQuickSwitcherModal.ts#L598

```ts
    if (!option.keepOpen) {
      this.close();
      this.navigate(() => this.isClosed); // wait for close to finish before navigating
    } else if (leaf === "same-tab") {
      this.stateToRestore ??= this.appHelper.captureState(this.initialLeaf);
    }

    this.navigate(() =>
      this.appHelper.openFile(
        fileToOpened,
        { leaf, offset, inplace: option.keepOpen },
        this.stateToRestore
      )
    );
```

In the `this.close()` 

```ts
   e.prototype.close = function() {
      // ...
      Lt.isPhone ? function(e, t) {
          return new Promise((function(n) {
            // ...
          ))
      }(o, a).then((function() {
          r.detach()
          e.onClose()
      }
      )) : (r.detach(),
      this.onClose()),
```

When we use Obsidian on the mobile layout, `Lt.isPhone` becomes `true`, then it closes the dialog with animation as asynchronous operations. As a result, it wouldn't open the file correctly because it operates before the dialog is closed.

We can solve this for now by inserting sleep... but it's not elegant...

```diff
    if (!option.keepOpen) {
      this.close();
+     await sleep(500);
      this.navigate(() => this.isClosed); // wait for close to finish before navigating
    } else if (leaf === "same-tab") {
      this.stateToRestore ??= this.appHelper.captureState(this.initialLeaf);
    }
```

**Comment by tadashi-aikawa** [comment]
This is not a perfect solution, but it is the safest and simplest fix I can think of now.

```diff
    if (!option.keepOpen) {
+     if (Platform.isMobile) {
+       // https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/issues/207
+       this.onClose();
+     }
      this.close();
      this.navigate(() => this.isClosed); // wait for close to finish before navigating
    } else if (leaf === "same-tab") {
      this.stateToRestore ??= this.appHelper.captureState(this.initialLeaf);
    }

```

**Comment by tadashi-aikawa** [comment]
@YIRU69 @kometenstaub 
Fix in [v9.2.0](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/9.2.0) 🚀

**Comment by YIRU69** [comment]
Thank you! It  work now!
